src/java/org/apache/cassandra/cql3/statements/DescribeStatement.java

```diff
@@ -132,8 +132,13 @@ public final ResultMessage execute(QueryState state, QueryOptions options, long
     @Override
     public ResultMessage executeLocally(QueryState state, QueryOptions options)
     {
-        Keyspaces keyspaces = Schema.instance.distributedAndLocalKeyspaces();
-        UUID schemaVersion = Schema.instance.getVersion();
+        UUID schemaVersion;
+        Keyspaces keyspaces;
+        do
+        {
+            schemaVersion = Schema.instance.getVersionSync();
+            keyspaces = Schema.instance.distributedAndLocalKeyspaces();
```

Comment:
will the version always be inconsistent for a long time ,and we are in this loop for long time.

Reply:
Not sure - is it a statement or question? I think we can skip this sync if paging is not requested; if paging was requested and there happened what you say, we would never be able to return results as paging would always fail due to concurrent schema changes. On the other hand - I don't know under what circumstances that could happen?

Reply:
Actually, I don't think this is a case - it is not waiting here for the cluster schema to be consistent - it is only the local thing - when the schema gets transformed locally, the transformation is applied in a synchronized block. A single transformation may include multiple items which are applied one by one immediately updating the snapshot. On the other hand, scheme version is updated once after the whole transformation is completed. So, here we are just waiting for that synchronized block to complete and make sure that no other synchronized transformation has started in the meantime. In other words, waiting to make sure the metadata and version pertains to the same schema.

Reply:
Oh, I see.